hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/ha/OMHAMetrics.java

```diff
@@ -0,0 +1,148 @@
+
+/**
+ * Class to maintain metrics and info related to OM HA.
+ */
+@Metrics(about = "OzoneManager HA Metrics", context = OzoneConsts.OZONE)
+public final class OMHAMetrics implements MetricsSource {
+
+  private enum OMHAMetricsInfo implements MetricsInfo {
+
+    OzoneManagerHALeaderState("Leader active state " +
+        "of OzoneManager node (1 leader, 0 follower)"),
+    NodeId("OM node Id");
+
+    private final String description;
+
+    OMHAMetricsInfo(String description) {
+      this.description = description;
+    }
+
+    @Override
+    public String description() {
+      return description;
+    }
+  }
+
+  /**
+   * Private nested class to hold
+   * the values of OMHAMetricsInfo.
+   */
+  private static final class OMHAInfo {
+
+    private long ozoneManagerHALeaderState;
+    private String nodeId;
+
+    OMHAInfo() {
+      this.ozoneManagerHALeaderState = 0L;
+      this.nodeId = "";
+    }
+
+    public long getOzoneManagerHALeaderState() {
+      return ozoneManagerHALeaderState;
+    }
+
+    public void setOzoneManagerHALeaderState(long ozoneManagerHALeaderState) {
+      this.ozoneManagerHALeaderState = ozoneManagerHALeaderState;
+    }
+
+    public String getNodeId() {
+      return nodeId;
+    }
+
+    public void setNodeId(String nodeId) {
+      this.nodeId = nodeId;
+    }
+  }
+
+  public static final String SOURCE_NAME =
+      OMHAMetrics.class.getSimpleName();
+  private final OMHAInfo omhaInfo = new OMHAInfo();
+  private MetricsRegistry metricsRegistry;
+
+  private String currNodeId;
+  private String leaderId;
+
+  public OMHAMetrics(String currNodeId, String leaderId) {
+    this.currNodeId = currNodeId;
+    this.leaderId = leaderId;
+    this.metricsRegistry = new MetricsRegistry(SOURCE_NAME);
+  }
+
+  /**
+   * Create and return OMHAMetrics instance.
+   * @return OMHAMetrics
+   */
+  public static synchronized OMHAMetrics create(
+      String nodeId, String leaderId) {
+    OMHAMetrics metrics = new OMHAMetrics(nodeId, leaderId);
+    return DefaultMetricsSystem.instance()
+        .register(SOURCE_NAME, "Metrics for OM HA", metrics);
+  }
+
+  /**
+   * Unregister the metrics instance.
+   */
+  public static void unRegister() {
+    DefaultMetricsSystem.instance().unregisterSource(SOURCE_NAME);
+  }
+
+  @Override
+  public synchronized void getMetrics(MetricsCollector collector, boolean all) {
+
+    MetricsRecordBuilder recordBuilder = collector.addRecord(SOURCE_NAME);
+
+    if (currNodeId.equals(leaderId)) {
+      omhaInfo.setNodeId(currNodeId);
+      omhaInfo.setOzoneManagerHALeaderState(1);
+
+      recordBuilder
+          .tag(OMHAMetricsInfo.NodeId, currNodeId)
+          .addGauge(OMHAMetricsInfo.OzoneManagerHALeaderState, 1);
+    } else {
+      omhaInfo.setNodeId(currNodeId);
+      omhaInfo.setOzoneManagerHALeaderState(0);
+
+      recordBuilder
+          .tag(OMHAMetricsInfo.NodeId, currNodeId)
+          .addGauge(OMHAMetricsInfo.OzoneManagerHALeaderState, 0);
+    }
```

Comment:
Please reduce duplication.  The only difference between the branches is the leader state value (0 or 1).

```suggestion
    int state = currNodeId.equals(leaderId) ? 1 : 0;
    omhaInfo.setNodeId(currNodeId);
    omhaInfo.setOzoneManagerHALeaderState(state);

    recordBuilder
        .tag(OMHAMetricsInfo.NodeId, currNodeId)
        .addGauge(OMHAMetricsInfo.OzoneManagerHALeaderState, state);
```